src/test/java/executor/service/service/impl/ExecutionServiceImplTest.java

```diff
@@ -0,0 +1,47 @@
+package executor.service.service.impl;
+
+import executor.service.model.Scenario;
+import executor.service.service.ScenarioExecutor;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.openqa.selenium.WebDriver;
+
+import static org.mockito.Mockito.*;
+
+/**
+ * Class for testing the implementation of an ExecutionServiceImpl.class.
+ *
+ *  @author Oleksandr Tuleninov
+ *  @version 01
+ * */
+public class ExecutionServiceImplTest {
+
+    private ScenarioExecutor scenarioExecutor;
+    private Scenario scenario;
+    private WebDriver webDriver;
+
+    @BeforeEach
+    public void setUp() {
+        this.scenarioExecutor = mock(ScenarioExecutor.class);
+        this.scenario = mock(Scenario.class);
+        this.webDriver = mock(WebDriver.class);
+    }
+
+    @AfterEach
+    void tearDown() {
+        verifyNoMoreInteractions(scenarioExecutor);
+        this.scenarioExecutor = null;
+        this.scenario = null;
+        this.webDriver = null;
+    }
+
+    @Test
+    public void testExecute() {
+        doNothing().when(scenarioExecutor).execute(scenario, webDriver);
+
+        scenarioExecutor.execute(scenario, webDriver);
+
+        verify(scenarioExecutor, times(1)).execute(scenario, webDriver);
```

Comment:
What are you testing here?
ExecutionServiceImpl? ScenarioExecutor? Or Mockito.doNothing????

Reply:
Here we test the presence and call of the **scenarioExecutor.execute(scenario, webDriver);** method. in the ExecutionServiceImpl class. If the question raises the name of the test method, I can correct its name. If you don't understand how Mockito works, its verify() method can be found in the lecture.

Reply:
Here I see standalone call of **scenarioExecutor.execute(scenario, webDriver);**. It will not test the presence of this method in the *ExecutionServiceImpl* class. You should create instanse of *ExecutionServiceImpl* and call its method, previously mocked all the dependencies.

Reply:
I modified the script on a simple java object for testing.

Reply:
Look at *Mockito.any(Class)*. Try use this while mocking.
Actually you're trying to test **ExecutionServiceImpl** so please create instanse of **ExecutionServiceImpl**.  After that call its method *execute(scenario, proxy)* and *verify* that *scenarioExecutor.execute()* has been called.